pkg/rule/rule.go

```diff
@@ -213,6 +213,21 @@ func (m *Manager) Update(evalInterval time.Duration, files []string) error {
 			continue
 		}
 	}
+
+	for s, mgr := range m.mgrs {
```

Comment:
it might be worth adding a comment explaining that this section removes the rules from a manager when a strategy has no more rule.